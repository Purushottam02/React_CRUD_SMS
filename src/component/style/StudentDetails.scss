.box-wrapper {
  display: flex;
  justify-content: center;
  align-items: center;
  height: 100vh;
  flex-direction: column;
}
.details {
  display: grid;
  padding: 15px;
  background-color: #f0f0f0;
  border-radius: 8px;
  box-shadow: 0 2px 6px rgba(0, 0, 0, 0.15);
  max-width: 25%;
  width: 100%;
  .row {
    display: flex;
    justify-content: flex-start;
    padding: 8px 0px 10px 15px;
    font-size: 16px;

    span {
      &:first-child {
        font-weight: bold;
        width: 40%;
      }

      &.value {
        color: #424646;
        padding-left: 25px;
        &:hover {
          text-decoration: none;
        }
      }
    }
  }
}
