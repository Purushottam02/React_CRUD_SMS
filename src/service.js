import axios from "axios";
export async function getStudents() {
  const response = await axios.get("http://localhost:8000/students", {
    params: {},
  });
  return response.data;
}
export async function createStudents(student) {
  const response = await fetch("http://localhost:8000/students", {
    method: "POST",
    headers: {
      Accept: "application/json",
      "Content-Type": "application/json",
    },
    body: JSON.stringify(student),
  });
  const data = await response.json();
  return data;
}

export async function getStudentDetails(rollNo) {
  try {
    const response = await fetch(
      `http://localhost:8000/students/details/${rollNo}`,
      {
        method: "GET",
        headers: {
          Accept: "application/json",
          "Content-Type": "application/json",
        },
      }
    );

    if (!response.ok) {
      throw new Error("Error fetching student details");
    }

    const data = await response.json();
    return data;
  } catch (error) {
    console.error("Error:", error);
    throw error;
  }
}

export async function deleteStudent(studentId) {
  const response = await fetch(`http://localhost:8000/students/${studentId}`, {
    method: "DELETE",
    body: JSON.stringify(studentId),
  });
  const students = await response.text();
  return students;
}

export async function editStudentDetails(rollNo, formData) {
  try {
    const response = await fetch(
      `http://localhost:8000/students/edit/${rollNo}`,
      {
        method: "PUT",
        headers: {
          Accept: "application/json",
          "Content-Type": "application/json",
        },
        body: JSON.stringify(formData),
      }
    );

    if (!response.ok) {
      throw new Error("Error updating form data");
    }

    console.log("Form data updated successfully!");
  } catch (error) {
    console.error("Error:", error);
  }
}
