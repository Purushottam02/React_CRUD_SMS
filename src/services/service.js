export async function getStudents() {
  try {
    const response = await fetch("http://localhost:8000/students", {
      method: "GET",
    });
    if (response.ok) {
      const data = await response.json();
      return data;
    } else {
      throw new Error("Failed to fetch students data");
    }
  } catch (error) {
    console.error("Error:", error);
    throw error;
  }
}

export async function createStudents(student) {
  const response = await fetch("http://localhost:8000/students", {
    method: "POST",
    headers: {
      Accept: "application/json",
      "Content-Type": "application/json",
    },
    body: JSON.stringify(student),
  });
  const data = await response.json();
  return data;
}

export async function getStudentDetails(rollNo) {
  try {
    const response = await fetch(
      `http://localhost:8000/students/details/${rollNo}`,
      {
        method: "GET",
        headers: {
          Accept: "application/json",
          "Content-Type": "application/json",
        },
      }
    );

    if (!response.ok) {
      throw new Error("Error fetching student details");
    }

    const data = await response.json();
    return data;
  } catch (error) {
    console.error("Error:", error);
    throw error;
  }
}

export async function deleteStudent(studentId) {
  const response = await fetch(`http://localhost:8000/students/${studentId}`, {
    method: "DELETE",
    body: JSON.stringify(studentId),
  });
  const students = await response.text();
  return students;
}

export async function editStudentDetails(rollNo, formData) {
  try {
    const response = await fetch(
      `http://localhost:8000/students/edit/${rollNo}`,
      {
        method: "PUT",
        headers: {
          Accept: "application/json",
          "Content-Type": "application/json",
        },
        body: JSON.stringify(formData),
      }
    );

    if (!response.ok) {
      throw new Error("Error updating form data");
    }

    console.log("Form data updated successfully!");
  } catch (error) {
    console.error("Error:", error);
  }
}
