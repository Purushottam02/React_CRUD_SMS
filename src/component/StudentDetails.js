import "./style/StudentDetails.scss";
import { useEffect, useState } from "react";
import { getStudentDetails } from "../service";
import { useParams } from "react-router-dom";
import { HEDER_TEXT } from "../constant";

function StudentDetails() {
  const [students, setStudents] = useState();
  const { rollNo } = useParams();

  useEffect(() => {
    const init = async () => {
      const student = await getStudentDetails(rollNo);
      setStudents(student);
    };

    init();
  }, [rollNo]);

  return (
    <div className="box-wrapper">
      <div className="details">
        {students &&
          Object.keys(HEDER_TEXT).map((field) => {
            if (field !== "edit" && field !== "delete") {
              return (
                <div key={field} className="row">
                  <span>{HEDER_TEXT[field]}:</span>
                  <span className="value">{students[field]}</span>
                </div>
              );
            }
            return null;
          })}
      </div>
    </div>
  );
}

export default StudentDetails;
