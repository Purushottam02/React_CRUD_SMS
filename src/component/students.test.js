import React from "react";
import { render, waitFor, screen } from "@testing-library/react";
import StudentDetails from "./StudentDetails";
import { getStudentDetails } from "../service";

jest.mock("../service");

test("renders StudentDetails component with student details", async () => {
  const mockStudent = {
    rollNo: "123",
    name: "John Doe",
    gender:"Male",
    physics:56,
    maths:56,
    english:67
  };

  getStudentDetails.mockResolvedValue(mockStudent);

  const app = render(<StudentDetails/>);

  // Wait for async tasks to complete
  await waitFor(() => expect(getStudentDetails).toHaveBeenCalledTimes(1));
  const rollNo = app.container.querySelector(".details");
  expect(rollNo).toBe("");
//   expect(screen.getByText("rollNo: 123")).toBeInTheDocument();
//   expect(screen.getByText("Name: John Doe")).toBeInTheDocument();
//   expect(screen.getByText("Age: 20")).toBeInTheDocument();
//   expect(screen.getByText("Grade: A")).toBeInTheDocument();
});
