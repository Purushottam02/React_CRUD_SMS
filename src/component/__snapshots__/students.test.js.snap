// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders StudentDetails component with student details 1`] = `[Function]`;

exports[`renders StudentDetails component with student details 2`] = `"name: John Doe"`;

exports[`renders StudentDetails component with student details 3`] = `"age: 20"`;

exports[`renders StudentDetails component with student details 4`] = `"grade: A"`;
